In [1]:
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns   
import pandas as pd
from scipy import stats

df = pd.read_csv('../../../Data/processed/cleaned_customer_data.csv')

Loyalty / relationship กับแบรนด์
1. Membership_Years
2. Payment_Method_Diversity
3. Wishlist_Items

In [2]:
Loyalty_features = [
    "Membership_Years",
    "Payment_Method_Diversity",
    "Wishlist_Items"
]

In [3]:
df[Loyalty_features].corrwith(df["Churned"]).sort_values()

Wishlist_Items             -0.189569
Membership_Years           -0.000623
Payment_Method_Diversity    0.004767
dtype: float64

Loyalty Feature Analysis

Loyalty-related features show very weak correlations with churn.
Wishlist items demonstrate a small negative relationship with churn (-0.19), suggesting that customers who actively save items may be slightly less likely to churn.
However, other loyalty indicators such as membership duration and payment method diversity exhibit almost no correlation with churn.



Summary:
Overall, loyalty features appear to have limited explanatory power compared to engagement and behavioral metrics.